jira/client.py

```diff
@@ -325,7 +325,7 @@ class JIRA(object):
     AGILE_BASE_URL = GreenHopperResource.AGILE_BASE_URL
 
     def __init__(self, server=None, options=None, basic_auth=None, oauth=None, jwt=None, kerberos=False, kerberos_options=None,
```

Comment:
One option would have been to accept kwargs and then check if `async` had been set.